From f6c23ad7fee948197a326816834895653a48b366 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Tue, 7 Feb 2017 14:10:54 +0800
Subject: [PATCH 1501/5242] MLK-13893 net: fec: do defer probe when mii bus
 not registered

commit  27f1f560702cf0f89cc9bfab12b20da83d160328 from
https://source.codeaurora.org/external/imx/linux-imx.git

When two MACs share one mii bus, MAC driver should do defer probe
when the mii bus not registered.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index daa9b14..03789f1 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2113,6 +2113,8 @@ static int fec_enet_mii_init(struct platform_device *pdev)
 	err = of_mdiobus_register(fep->mii_bus, node);
 	if (node)
 		of_node_put(node);
+	else if (fep->phy_node)
+		err = -EPROBE_DEFER;
 	if (err)
 		goto err_out_free_mdiobus;
 
@@ -3709,8 +3711,10 @@ static void fec_enet_of_parse_stop_mode(struct platform_device *pdev)
 
 	init_completion(&fep->mdio_done);
 	ret = fec_enet_mii_init(pdev);
-	if (ret)
+	if (ret) {
+		dev_id = 0;
 		goto failed_mii_init;
+	}
 
 	/* Carrier starts down, phylib will bring it up */
 	netif_carrier_off(ndev);
-- 
1.7.9.5

